.github/workflows/buildndeploy.yml

```
@@ -19,6 +22,14 @@ jobs:
     steps:
       - name: Checkout Repository
         uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Determine Workflow Branch
+        run: echo "TARGET_BRANCH=${{ github.event.pull_request.head.ref || github.ref_name }}" >> $GITHUB_ENV
```

actionlint

shellcheck reported issue in this script: SC2086:info:1:86: Double quote to prevent globbing and word splitting

(shellcheck)